src/runner.ts

```diff
@@ -228,15 +209,54 @@ export class TestsRunner {
         cliProcess.on("error", (err) => { reject(err); });
     }
 
-    private getJavaPath(javaRootPath: string): string | undefined {
-        if (!javaRootPath) {
+    private getSOAtestJavaPath(installDir: string): string | undefined {
```

Comment:
How about:
```
private getSOAtestJavaPath(installDir: string): string | undefined {
    if (!installDir) {
        const soatestcliPath = which.sync('soatestcli');
        installDir = soatestcliPath.substring(0, soatestcliPath.lastIndexOf('soatestcli') - 1);
    }
    if (!fs.existsSync(installDir)) {
        core.warning("Unable to process the XML report using Java bundled with SOAtest because the SOAtest installation directory is missing.");
        return;
    }
    let javaFilePath = this.doGetSOAtestJavaPath(installDir);
    if (!javaFilePath) {
        core.warning("Unable to process the XML report using Java bundled with SOAtest because it is missing.");
    } else {
        core.info(messagesFormatter.format(messages.found_java_at, javaFilePath));
    }
    return javaFilePath;
}
```
I think the logic looks more straightforward in this way. Please localize the messages.